specs/phase0/fork-choice.md

```diff
@@ -371,27 +498,13 @@ def update_latest_messages(store: Store, attesting_indices: Sequence[ValidatorIn
 
 ```python
 def on_tick(store: Store, time: uint64) -> None:
-    previous_slot = get_current_slot(store)
-
-    # update store time
-    store.time = time
-
-    current_slot = get_current_slot(store)
-
-    # Reset store.proposer_boost_root if this is a new slot
-    if current_slot > previous_slot:
-        store.proposer_boost_root = Root()
-
-    # Not a new epoch, return
-    if not (current_slot > previous_slot and compute_slots_since_epoch_start(current_slot) == 0):
-        return
-
-    # Update store.justified_checkpoint if a better checkpoint on the store.finalized_checkpoint chain
-    if store.best_justified_checkpoint.epoch > store.justified_checkpoint.epoch:
-        finalized_slot = compute_start_slot_at_epoch(store.finalized_checkpoint.epoch)
-        ancestor_at_finalized_slot = get_ancestor(store, store.best_justified_checkpoint.root, finalized_slot)
-        if ancestor_at_finalized_slot == store.finalized_checkpoint.root:
-            store.justified_checkpoint = store.best_justified_checkpoint
+    # If the ``store.time`` falls behind, catch up slot by slot to
+    # ensure that every previous slot will be processed with ``on_tick_per_slot``.
```

Comment:
I originally read this comment as though the caller needs to be careful to call for each slot, but the function catches up for you. I think the following is a little bit clearer

```suggestion
    # If the ``store.time`` falls behind, while loop catches up slot by slot
    # to ensure that every previous slot is processed with ``on_tick_per_slot``
```